include/ginkgo/core/matrix/csr.hpp

```diff
@@ -177,13 +177,39 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
         /**
          * Creates a classical strategy.
          */
-        classical() : strategy_type("classical") {}
+        classical() : strategy_type("classical"), max_length_per_row_(0) {}
 
         void process(const Array<index_type> &mtx_row_ptrs,
                      Array<index_type> *mtx_srow) override
-        {}
+        {
+            auto host_mtx_exec = mtx_row_ptrs.get_executor()->get_master();
+            Array<index_type> row_ptrs_host(host_mtx_exec);
+            const bool is_mtx_on_host{host_mtx_exec ==
+                                      mtx_row_ptrs.get_executor()};
+            const index_type *row_ptrs{};
+            if (is_mtx_on_host) {
+                row_ptrs = mtx_row_ptrs.get_const_data();
+            } else {
+                row_ptrs_host = mtx_row_ptrs;
+                row_ptrs = row_ptrs_host.get_const_data();
+            }
+            auto num_rows = mtx_row_ptrs.get_num_elems() - 1;
+            max_length_per_row_ = 0;
+            for (index_type i = 1; i < num_rows + 1; i++) {
+                max_length_per_row_ = std::max(max_length_per_row_,
+                                               row_ptrs[i] - row_ptrs[i - 1]);
+            }
+        }
 
         int64_t clac_size(const int64_t nnz) override { return 0; }
```

Comment:
I think this is supposed to be `calc_size` ?

Reply:
Yes, It should be.
To keep the same public interface, I can not change it.

Reply:
Ah, that is unfortunate. Do we have someplace where me mention the things we need to update for the next version 2.0 which will break the public interface ? If not, can you create an issue and update this and also the stuff mentioned in #320 there ?